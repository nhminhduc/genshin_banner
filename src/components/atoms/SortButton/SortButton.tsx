import cx from "classnames";
import { ComponentProps, ReactElement } from "react";

type SortButtonProp = {
  className?: string;
  icon?: ReactElement<ComponentProps<"svg" | "img">>;
  id: string;
  label?: string;
  name: string;
  onButtonClick(isChecked: boolean): void;
  sortDesc?: boolean;
};

const SortButton = ({
  className,
  icon,
  id,
  label,
  name,
  onButtonClick,
  sortDesc = false,
}: SortButtonProp) => {
  const handleOnClick = () => {
    onButtonClick(!sortDesc);
  };

  const Icon = icon ?? null;
  return (
    <button
      className={cx("flex items-center", sortDesc, className, {
        "opacity-30": !sortDesc,
      })}
      id={id}
      name={name}
      onClick={handleOnClick}
      type="button"
    >
      {Icon} {label}
    </button>
  );
};

export default SortButton;
