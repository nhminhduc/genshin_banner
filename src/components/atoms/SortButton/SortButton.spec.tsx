import { render } from "@testing-library/react";
import { describe, expect, test, vi } from "vitest";

import SortButton from "./SortButton";

describe("SortButton test", () => {
  test("SortButton load", () => {
    const mockFunction = vi.fn();
    const { container } = render(
      <SortButton
        id="SortButton-test"
        name="SortButton test"
        onButtonClick={mockFunction}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
