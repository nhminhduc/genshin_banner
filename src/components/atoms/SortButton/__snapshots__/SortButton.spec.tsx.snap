// Vitest Snapshot v1

exports[`SortButton test > SortButton load 1`] = `
<div>
  <button
    class="flex items-center opacity-30"
    id="SortButton-test"
    name="SortButton test"
    type="button"
  >
     
  </button>
</div>
`;
